Library/Homebrew/tap.rb

```diff
@@ -722,7 +722,11 @@ def formula_renames
   # Hash with tap migrations.
   sig { returns(Hash) }
   def tap_migrations
-    @tap_migrations ||= if (migration_file = path/HOMEBREW_TAP_MIGRATIONS_FILE).file?
+    @tap_migrations ||= if name == "homebrew/cask" && !Homebrew::EnvConfig.no_install_from_api?
```

Comment:
This is the same approach taken for `def cask_renames` above and is part of the motivation for https://github.com/Homebrew/brew/issues/15597 (but shouldn't block this PR IMO).